### PR TITLE
[BT] add `BetterTransformer` support for ViLT architecture

### PR DESCRIPTION
**Pull request by ka00ri**
# What does this PR do?

Added BetterTransformer support for ViLT architecture. 
Added test model for ViltModel.

Tested the conversion of ViltLayer:

```
from optimum.bettertransformer import BetterTransformer
model = BetterTransformer.transform("hf-internal-testing/tiny-random-ViltModel")
print(model)
ViltModel(
  (embeddings): ViltEmbeddings(
    (text_embeddings): TextEmbeddings(
      (word_embeddings): Embedding(1124, 32)
      (position_embeddings): Embedding(512, 32)
      (token_type_embeddings): Embedding(16, 32)
      (LayerNorm): LayerNorm((32,), eps=1e-12, elementwise_affine=True)
      (dropout): Dropout(p=0.1, inplace=False)
    )
    (patch_embeddings): ViltPatchEmbeddings(
      (projection): Conv2d(3, 32, kernel_size=(2, 2), stride=(2, 2))
    )
    (token_type_embeddings): Embedding(2, 32)
    (dropout): Dropout(p=0.1, inplace=False)
  )
  (encoder): ViltEncoder(
    (layer): ModuleList(
      (0): ViltLayerBetterTransformer()
      (1): ViltLayerBetterTransformer()
      (2): ViltLayerBetterTransformer()
      (3): ViltLayerBetterTransformer()
      (4): ViltLayerBetterTransformer()
    )
  )
  (layernorm): LayerNorm((32,), eps=1e-12, elementwise_affine=True)
  (pooler): ViltPooler(
    (dense): Linear(in_features=32, out_features=32, bias=True)
    (activation): Tanh()
  )
)

```

To: @younesbelkada @michaelbenayoun  @fxmarty

### CODE REVIEW

**Review by younesbelkada — COMMENTED** [review]
Very nice! Thank you very much for adding `BetterTransformer` support for `ViLT` model! 
I think we need 2 more additional steps before merging:
1- Could you add a new class on the testing suite since the model expected text + image. So inside `tests_bettertransformer_vision.py` you can just add something like: 
```
ALL_VISION_TEXT_MODELS_TO_TEST = ("hf-internal-testing/tiny-random-ViltModel")

class BetterTransformersViLTTest(BetterTransformersTestMixin, unittest.TestCase):
    r"""
    Testing suite for Vision Models - tests all the tests defined in `BetterTransformersTestMixin`
    """
    all_models_to_test = ALL_VISION_TEXT_MODELS_TO_TEST

    def prepare_inputs_for_class(self, model_id=None):
        url = "http://images.cocodataset.org/val2017/000000039769.jpg"
        image = Image.open(requests.get(url, stream=True).raw)
        processor = AutoProcessor.from_pretrained("hf-internal-testing/tiny-random-ViltModel")
        .... 
        return inputs
```
And tweak the function `prepare_inputs_for_class` based on this example: https://huggingface.co/dandelin/vilt-b32-finetuned-vqa

2- Can you add the new architecture on the documentation? 🙏 this would require adding just a line [here](https://github.com/huggingface/optimum/blob/889df93be26df5868a2978c9a0031c6043d12913/docs/source/bettertransformer/overview.mdx?plain=1#L47) (don't forget to respect the alphabetical order ;) )

Also don't forget to run `make style` before pushing! ;) 

Thanks a bunch!

File: tests/bettertransformer/test_bettertransformer_vision.py [hunk 1]
```diff
@@ -30,6 +30,11 @@
 ]
 
 
+ALL_VISION_TEXT_MODELS_TO_TEST = [
+    "hf-internal-testing/tiny-random-ViltModel",
```
Comment (younesbelkada) [inline tests/bettertransformer/test_bettertransformer_vision.py:34]:
```suggestion
    "hf-internal-testing/tiny-vilt-random-vqa",
```
I think the previous model is broken, let's try with this one!

Comment (ka00ri) [reply tests/bettertransformer/test_bettertransformer_vision.py]: Thanks for the feedback. I did fix the alphabetic order and added the test for vision and text models, but the test still didn't pass.

Comment (younesbelkada) [reply tests/bettertransformer/test_bettertransformer_vision.py]: Yeah I saw that! Thanks a lot for fixing the alphabetical order ;) ! let's see now if the test passes with the new model

File: tests/bettertransformer/test_bettertransformer_vision.py [hunk 2]
```diff
@@ -44,3 +49,20 @@ def prepare_inputs_for_class(self, model_id=None):
...
+    """
+    all_models_to_test = ALL_VISION_TEXT_MODELS_TO_TEST
+
+    def prepare_inputs_for_class(self, model_id=None):
+        url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+        image = Image.open(requests.get(url, stream=True).raw)
+        text = "How many cats are there?"
+
+        # Model takes image and text as input
+        processor = AutoProcessor.from_pretrained("hf-internal-testing/tiny-random-ViltModel")
```
Comment (younesbelkada) [inline tests/bettertransformer/test_bettertransformer_vision.py:66]:
```suggestion
        processor = AutoProcessor.from_pretrained(model_id)
```
We forgot to add this @ka00ri ! 💪 Now the tests should pass

Comment (ka00ri) [reply tests/bettertransformer/test_bettertransformer_vision.py]: Good catch!  let's see🤞

Comment (ka00ri) [reply tests/bettertransformer/test_bettertransformer_vision.py]: The current fail seems to be due to a missing file...I am not sure if I can fix this.

**Review by younesbelkada — APPROVED** [review]
Very clean and super fast implementation of `ViLT` support for `BetterTransformer`! Thanks also for taking care of slightly refactoring the documentation.
Thanks a lot for adding this, you really did a great job here! 💪 
I have nothing to add on my side, gently pinging @fxmarty and @michaelbenayoun for a last review!

**Pull request merged**